### PR TITLE
New plugin architecture

### LINKED ISSUE #2611: PluginLoaders not found for platform specific plugins
MvxAndroidSetup and the other platform setups look for a specific extension to the plugin assembly. So for Android it expects an assembly that is called `MyPlugin.Useful.Droid` where the Android implementation resides. If it doesn't find that, it doesn't try to fall back to anything else and then just explodes.

This happens due to when switching to Multi-target, the assembly names are now all the same for all assemblies. Platform specific or not.

@AlexStefan was very quick to provide a repro sample https://github.com/AlexStefan/template-app/blob/master/Internationalization/

The Core and Droid projects there point at MvvmCross in the folder `../../mc`

### Steps to reproduce :scroll:

1. Add one of the official plugins
2. Make sure Bootstrap is added
3. Load a ViewModel which uses that plugin
4. 💥 app crashes

### Expected behavior :thinking:
App should run

### Actual behavior :bug:
App crashes

### Configuration :wrench:

**Version:** 6.0.0-beta2

### PR DESCRIPTION
**Pull request by willsb**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
New plugin registering mechanism

### :arrow_heading_down: What is the current behavior?
Plugins need a file to be added when they are installed. But NuGet now doesn't add those files automagically.

### :new: What is the new behavior (if this is a feature change)?
Attribute registration, yay
Also, this burns useless classes, like the PluginLoader, which was just a plugin with extra steps

### :boom: Does this PR introduce a breaking change?
YES

### :bug: Recommendations for testing
Making sure the plugins are registered when using them in actual apps

### 📝 Links to relevant issues/docs 
Closes #2611

### :thinking: Checklist before submitting

- [X] All projects build
- [X] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [X] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide)) 
- [X] Rebased onto current develop

### CODE REVIEW

**Comment by Cheesebaron** [comment]
I'd like to see some unit tests covering this.

**Comment by Cheesebaron** [comment]
Good stuff @willsb

File: MvvmCross.Plugins/MethodBinding/PluginLoader.cs [hunk 1]
```diff
@@ -3,24 +3,18 @@
 // See the LICENSE file in the project root for more information.
```
Comment (nmilcoff) [inline MvvmCross.Plugins/MethodBinding/PluginLoader.cs]: File should be renamed `PluginLoader` -> `Plugin`

File: MvvmCross.Plugins/Visibility/Platform/Uap/PlugIn.cs [hunk 2]
```diff
@@ -2,14 +2,12 @@
 // The .NET Foundation licenses this file to you under the MS-PL license.
```
Comment (nmilcoff) [inline MvvmCross.Plugins/Visibility/Platform/Uap/PlugIn.cs]: File should be renamed `PlugIn` -> `Plugin`

File: MvvmCross.Plugins/Accelerometer/Platform/Uap/Plugin.cs [hunk 3]
```diff
@@ -2,13 +2,9 @@
...
 // See the LICENSE file in the project root for more information.
 
-
-using MvvmCross.Plugins;
-
 namespace MvvmCross.Plugin.Accelerometer.Platform.Uap
 {
-    public class Plugin
-        : IMvxPlugin
+    public class Plugin : IMvxPlugin
```
Comment (nmilcoff) [inline MvvmCross.Plugins/Accelerometer/Platform/Uap/Plugin.cs]: This class is missing the `MvxPlugin` attribute

File: MvvmCross.Plugins/Accelerometer/Platform/Wpf/Plugin.cs [hunk 4]
```diff
@@ -2,13 +2,9 @@
...
 // See the LICENSE file in the project root for more information.
 
-
-using MvvmCross.Plugins;
-
 namespace MvvmCross.Plugin.Accelerometer.Platform.Wpf
 {
-    public class Plugin
-        : IMvxPlugin
+    public class Plugin : IMvxPlugin
```
Comment (nmilcoff) [inline MvvmCross.Plugins/Accelerometer/Platform/Wpf/Plugin.cs]: This class is missing the `MvxPlugin` attribute

File: MvvmCross.Plugins/Color/Platform/Uap/Plugin.cs [hunk 5]
```diff
@@ -2,14 +2,11 @@
...
 
-
-using MvvmCross.Plugins;
 using MvvmCross.UI;
 
 namespace MvvmCross.Plugin.Color.Platform.Uap
 {
-    public class Plugin
-        : IMvxPlugin
+    public class Plugin : IMvxPlugin
```
Comment (nmilcoff) [inline MvvmCross.Plugins/Color/Platform/Uap/Plugin.cs]: This class is missing the `MvxPlugin` attribute

File: MvvmCross.Plugins/Color/Platform/Wpf/Plugin.cs [hunk 6]
```diff
@@ -6,13 +6,11 @@
...
 
 using MvvmCross.Platform;
-using MvvmCross.Platform.Plugins;
 using MvvmCross.Platform.UI;
 
 namespace MvvmCross.Plugin.Color.Platform.Wpf
 {
-    public class Plugin
-        : IMvxPlugin
+    public class Plugin : IMvxPlugin
```
Comment (nmilcoff) [inline MvvmCross.Plugins/Color/Platform/Wpf/Plugin.cs]: This class is missing the `MvxPlugin` attribute

File: MvvmCross.Plugins/Color/Plugin.cs [hunk 7]
```diff
@@ -3,21 +3,15 @@
...
 using MvvmCross.Converters;
-using MvvmCross.Plugins;
 
 namespace MvvmCross.Plugin.Color
 {
+    [MvxPlugin]
     [Preserve(AllMembers = true)]
-    public class PluginLoader
-        : IMvxPluginLoader
+    public sealed class PluginLoader : IMvxPlugin
```
Comment (nmilcoff) [inline MvvmCross.Plugins/Color/Plugin.cs]: This class isn't a platform implementation and it has a `MvxPlugin` attribute. Is that correct? Also seems like the class should be renamed to `Plugin` maybe?

Comment (willsb) [reply MvvmCross.Plugins/Color/Plugin.cs]: Yes, this is correct. The PluginLoader class was pretty much useless, so we are just registering plugins instead

**Comment by willsb** [comment]
@nmilcoff You comments were addressed :)

**Pull request merged**